tmt/steps/report/junit.py

```diff
@@ -58,40 +67,230 @@ def duration_to_seconds(duration: Optional[str]) -> Optional[int]:
             f"Malformed duration '{duration}' ({error}).")
 
 
+class ResultWrapper:
+    """
+    Adds possibility to wrap the ``tmt.Result`` and dynamically add more attributes which get
+    available inside the template context.
+    """
+
+    class PropertyDict(TypedDict):
+        """ Defines a property dict, which gets propagated into the final template properties. """
+
+        name: str
+        value: str
+
+    def __init__(self, wrapped: tmt.Result) -> None:
+        self._wrapped = wrapped
+        self._properties: dict[str, str] = {}
+
+    @property
+    def properties(self) -> list[PropertyDict]:
+        return [{'name': k, 'value': v} for k, v in self._properties.items()]
+
+    def set_properties(self, keyval: dict[str, str]) -> None:
+        self._properties = keyval
+
+    def __getattr__(self, name: str) -> Any:
+        """ Returns an attribute of a wrapped ``tmt.Result`` instance """
+        return getattr(self._wrapped, name)
+
+
+def _add_result_properties(result_instance: tmt.Result) -> ResultWrapper:
```

Comment:
* this function could be a class method of `ResultWrapper`, it's basically piece of its functionality, isolated outside of the class, and
* maybe it's just my impression, but is there any actuall difference between `_add_result_properties(result)` and `ResultWrapper(result)`? If not, I'd drop this function completely, it does not seem to have added value.

Reply:
> If not, I'd drop this function completely, it does not seem to have added value.

Correct, It's not anymore. Removed, thanks.